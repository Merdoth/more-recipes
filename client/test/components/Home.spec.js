import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedHome, { Home } from '../../components/Home.jsx';
import mockData from '../__mocks__/mockData';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


const setup = () => {
  const props = {
    recipes: [
      {
        id: 1,
        userId: 1,
        recipeName: 'Sweet cake',
        description: 'sweet',
        ingredients: 'floor, sugar, butter, salt',
        preparation: 'Mix and bake',
        image: '/img/toast_qaxxcy.jpg',
        views: 0,
        upVotes: 0,
        downVotes: 0,
        createdAt: '2018-02-13T22:14:54.283Z',
        updatedAt: '2018-02-18T15:23:59.106Z',
      }
    ],
    getMostVoted: () => { }
  };
  return shallow(<Home {...props} />);
};

describe('Home Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should return all recipes', () => {
    const wrapper = setup();
    wrapper.setState({
      recipes: [
        {
          id: 1,
          userId: 1
        }
      ]
    });
  });
});

describe('Connected Home component', () => {
  it('tests that the component successfully rendered', () => {
    const { recipe } = mockData;
    const store = mockStore({
      recipesReducer: {
        recipesFound: recipe.recipesFound
      }
    });
    const wrapper = shallow(<ConnectedHome store={store} />);
    expect(wrapper.length).toBe(1);
  });
});

