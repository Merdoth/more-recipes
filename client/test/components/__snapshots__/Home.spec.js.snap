// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component snapshot it should render the right amount of elements 1`] = `
<div>
  <Slider />
  <div
    className="recipe-wrapper"
  >
    <div
      className="container top"
    >
      <h6
        id="title4"
      >
        Top Recipes
      </h6>
      <div
        className="row"
      >
        <RecipeCard
          key="recipes-1"
          recipeList={
            Object {
              "createdAt": "2018-02-13T22:14:54.283Z",
              "description": "sweet",
              "downVotes": 0,
              "id": 1,
              "image": "/img/toast_qaxxcy.jpg",
              "ingredients": "floor, sugar, butter, salt",
              "preparation": "Mix and bake",
              "recipeName": "Sweet cake",
              "upVotes": 0,
              "updatedAt": "2018-02-18T15:23:59.106Z",
              "userId": 1,
              "views": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
